datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -187,6 +188,15 @@ public SdxCluster createSdx(final String userCrn, final String name, final SdxCl
         return sdxCluster;
     }
 
+    private void updateStackV4RequestWithEnvironmentCrnIfNotExistsOnIt(StackV4Request request, String environmentName) {
+        if (request != null && StringUtils.isEmpty(request.getEnvironmentCrn())) {
+            LOGGER.info("Internal Sdx request's stack request has no environment crn. Going to fetch it from the environment service...");
+            String crn = getEnvironment(environmentName).getCrn();
```

Comment:
As we discussed, this is a duplicated call

Reply:
done